src/main/java/com/waytta/clientinterface/BasicClient.java

```diff
@@ -1,88 +1,32 @@
 package com.waytta.clientinterface;
 
-public class BasicClient {
+import hudson.ExtensionPoint;
+import hudson.model.Describable;
+import hudson.model.Descriptor;
+import jenkins.model.Jenkins;
+import org.kohsuke.stapler.DataBoundConstructor;
+import hudson.DescriptorExtensionList;
+import java.util.List;
 
-    public static final int DEFAULT_JOB_POLL_TIME = 10;
-    private final String credentialsId;
-    private String target;
-    private String targetType;
-    private final String function;
-    private String batchSize = "100%";
-    private String mods = "";
-    private String pillarvalue = "";
-    private Boolean blockBuild = Boolean.FALSE;
-    private Integer jobPollTime = DEFAULT_JOB_POLL_TIME;
-    
-
-    public BasicClient(String credentialsId, String target, String targetType, String function) {
-        this.credentialsId = credentialsId;
-        this.target = target;
-        this.targetType = targetType;
-        this.function = function;
-    }
-
-
-    public String getPillarValue() {
-        return pillarvalue;
-    }
-
-    public void setPillarValue(String pillarvalue) {
-        this.pillarvalue = pillarvalue;
-    }
-
-    public String getMods() {
-        return mods;
-    }
-
-    public void setMods(String mods) {
-        this.mods = mods;
-    }
-
-    public String getTarget() {
-        return target;
-    }
-    
-    public void setTarget(String target) {
-    	this.target = target;
-    }
 
-    public String getTargetType() {
-        return targetType;
+public class BasicClient implements ExtensionPoint, Describable<BasicClient> {
+	public Descriptor<BasicClient> getDescriptor() {
```

Comment:
Just extend `AbstractDescribableImpl`, but `@Override` this method to cast the `super` value to `BasicClientDescriptor`.